org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -218,11 +218,13 @@ private String create() throws Exception {
 						+ " INNER JOIN M_Product p ON (p.M_Product_ID=po.M_Product_ID)"
 						+ " INNER JOIN M_DiscountSchemaLine dl ON (dl.M_DiscountSchemaLine_ID=?) "	//	#2
 						+ "WHERE p.AD_Client_ID IN (?, 0)"		//	#3
-						+ " AND p.IsActive='Y' AND po.IsActive='Y' AND po.IsCurrentVendor='Y'"
+						+ " AND p.IsActive='Y' AND po.IsActive='Y'"
 						//	Optional Restrictions
 						+ " AND (dl.Group1 IS NULL OR p.Group1=dl.Group1)"
 						+ " AND (dl.Group2 IS NULL OR p.Group2=dl.Group2)"
-						+ " AND (dl.C_BPartner_ID IS NULL OR po.C_BPartner_ID=dl.C_BPartner_ID)"
+						+ " AND (dl.VendorCategory IS NULL OR po.VendorCategory=dl.VendorCategory)"
+						+ " AND ((dl.C_BPartner_ID IS NULL AND po.IsCurrentVendor='Y')"
+						+ "      OR (po.C_BPartner_ID=dl.C_BPartner_ID AND (dl.IsIgnoreIsCurrentVendor='Y' OR po.IsCurrentVendor='Y')))"

```

Comment:
I think it should be just AND (dl.IsIgnoreIsCurrentVendor='Y' OR po.IsCurrentVendor='Y')

Reply:
But that would be an incompatible change. Currently you can build a pricelist which contains prices only for a given bpartner but only those prices where this bpartner is the currentvendor. I don't know whether this will break someones pricelists, but I think it isn't necessary to break compatibility if it could be easily maintained.

Reply:
why is this break the behaviour you mention above ? when dl.IsIgnoreIsCurrentVendor='N', it is checking whether po.IsCurrentVendor='Y' and that should be compatible with existing behaviour.

Reply:
I assume you meant to replace lines 226 and 227 by AND (dl.IsIgnoreIsCurrentVendor='Y' OR po.IsCurrentVendor='Y'). If not, i missunderstood you. Then given a dl.C_BPartner_ID is set products will be included, where there is no po line for the required vendor. Currently they won't be included.

Reply:
yes, replace 226 and 227 and keep line 225 above.

Reply:
Ok, now I remember why I did it this way. The user interface hides 'IsIgnoreIsCurrentVendor' as long as there is no Vendor given and it is set to 'N' by default. But of course it would be possible to set it to 'Y' even if no vendor is set. So I wanted to ignore it in this case.

But as we are only selecting the products and there is always on po-record with 'IsCurrentVendor' the product would be selected by this record anyway, it doesn't help to ignore others.

I've changed it.

Reply:
I just see I had an error in the above. If IsIgnoreIsCurrentVendor is set by accident (and not visible to user) and the VendorCategory is set, then products will be included where IsCurrentVendor is not set for the po-record with the correct VendorCategory. So products are included which are no want to be included. Of course it shouldn't happen that IsIgnoreIsCurrentVendor is set invisible but I personally would prefer to take it right then.

Reply:
With the displaylogic of "@DiscountType@=P & @C_BPartner_ID@!0" for "IsIgnoreIsCurrentVendor", we have to make sure "IsIgnoreIsCurrentVendor" is set to "N" when it is hidden due to the display logic above (perhaps in the beforesave event of the model class). It is easy to have unwanted side effect with a hidden IsIgnoreIsCurrentVendor that's set to 'Y' in db, best to avoid that.

Reply:
Ok, to make sure what to do: I have to regenerate X_M_DiscountSchemaLine and I_M_DiscountSchemaLine, add a beforeSave() function in MDiscountSchemaLine and fill IsIgnoreCurrentVendor with 'N' if C_BPartner_ID is 0. Right?

Reply:
yes, that should work. to test - 1) set IsIgnoreCurrentVendor=Y, C_BPartner_ID>0, save.  2) clear C_BPartner_ID to 0, save 3). check db, should have IsIgnoreCurrentVendor=N

Reply:
@hengsin, @CarlosRuiz-globalqss, 
I have added and tested setting IsIgnoreIsCurrentVendor to false (or N) if no businesspartner is given. I think, now all amendments are processed.
Regards,
Martin